logic/src/main/nl/jft/logic/tournament/Tournament.java

```diff
@@ -1,8 +1,187 @@
 package nl.jft.logic.tournament;
 
+import nl.jft.logic.participant.Participant;
+import nl.jft.logic.participant.ParticipantType;
+import nl.jft.logic.util.ParticipantUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
 /**
+ * A {@code Tournament} schedules {@link nl.jft.logic.match.Match matches} for the joined {@link Participant participants}.
+ * At the end of a {@code Tournament}, a winner is determined (depending on the {@link TournamentType}).
+ *
  * @author Lesley
  */
 public class Tournament {
 
+    private final List<Participant> participants = new ArrayList<>();
+
+    private final TournamentType tournamentType;
+    private final ParticipantType participantType;
+
+    private TournamentStatus status;
+    private Optional<Participant> winner;
+
+    /**
+     * Initializes a new {@code Tournament} using the given arguments.
+     *
+     * @param tournamentType  The {@code TournamentType} determines how {@link nl.jft.logic.match.Match matches}
+     *                        should be scheduled, and how a winner should be determined. Should not be {@code null}.
+     * @param participantType The {@code ParticipantType} determines what kind of {@link Participant participants}
+     *                        can join, should not be {@code null}.
+     * @throws NullPointerException If the given {@code TournamentType} is {@code null}.
+     * @throws NullPointerException If the given {@code ParticipantType} is {@code null}.
+     */
+    public Tournament(TournamentType tournamentType, ParticipantType participantType) {
+        this.tournamentType = Objects.requireNonNull(tournamentType);
+        this.participantType = Objects.requireNonNull(participantType);
+
+        status = TournamentStatus.SETUP;
+        winner = Optional.empty();
+    }
+
+    /**
+     * Starts this {@code Tournament}, given that it is still in the {@link TournamentStatus#SETUP} state.
+     *
+     * @throws IllegalStateException If this {@code Tournament} is in the {@link TournamentStatus#IN_PROGRESS} state.
+     * @throws IllegalStateException If this {@code Tournament} is in the {@link TournamentStatus#FINISHED} state.
+     */
+    public void start() {
+        if (status == TournamentStatus.IN_PROGRESS) {
+            throw new IllegalStateException("This tournament has already been started.");
+        }
+
+        if (status == TournamentStatus.FINISHED) {
+            throw new IllegalStateException("This tournament has already finished.");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "This tournament has already finished." 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://92.110.24.119:9000/coding_rules#rule_key=squid%3AS1192)